### PR TITLE
Feature/article management

### PR DESCRIPTION
**Pull request by kbmanisan**

### CODE REVIEW

File: database/migrations/2017_08_04_055318_article_table.php [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+    /**
+     * Run the migrations.
+     *
+     * @return void
+     */
+    public function up()
+    {
+        Schema::create('articles', function (Blueprint $table) {
+            $table->increments('id');
+            $table->integer('article_category_id')->unsigned();
```
Comment (inoue0204) [inline database/migrations/2017_08_04_055318_article_table.php]: $table->integer('article_category_id')->unsigned();
↓
$table->integer('article_category_id')->unsigned()->index();

File: app/Http/Controllers/Article/ArticleController.php [hunk 2]
```diff
@@ -2,30 +2,224 @@
...
+            'log' => Auth::user()->username. ' deleted article '. $article->title
+        ));
+        return redirect('articles');
+    }
+
+    public function validator($request)
+    {
+        return Validator::make($request->all(), [
+            'title' => 'required|max:255|unique:articles,title,NULL,id,deleted_at,NULL',
+            'article_category_id' => 'required|exists:categories,id,id,'.$request->all()['article_category_id'],
```
Comment (inoue0204) [inline app/Http/Controllers/Article/ArticleController.php]: 'article_category_id' => 'required|exists:categories,id,id,'.$request->all()['article_category_id'],
↓
'article_category_id' => 'required|exists:categories,id,deleted_at,NULL',

File: app/Http/Controllers/Category/CategoryController.php [hunk 3]
```diff
@@ -2,30 +2,201 @@
 
 namespace App\Http\Controllers\Category;
 
-use App\Http\Requests;
 use Illuminate\Http\Request;
+
+use App\Http\Requests;
 use App\Http\Controllers\Controller;
+use App\Modules\CategoryModule as CategoryModule;
```
Comment (inoue0204) [inline app/Http/Controllers/Category/CategoryController.php]: remove "as CategoryModule","  as ArticleModule", "as LogModule"

File: tests/AddArticle.php [hunk 4]
```diff
@@ -1,8 +1,6 @@
 <?php
 
 use Illuminate\Foundation\Testing\WithoutMiddleware;
-use Illuminate\Foundation\Testing\DatabaseMigrations;
-use Illuminate\Foundation\Testing\DatabaseTransactions;
 use App\User;
 
 class AddArticle extends TestCase
```
Comment (johndev182) [inline tests/AddArticle.php]: For Unit Tests, It's better to use one UnitTest per Object.. like ArticleTest, UserTest, etc.

**Pull request merged**